src/components/timeline/TimelineMarkerCanvas.js

```diff
@@ -232,8 +228,12 @@ class TimelineMarkerCanvas extends PureComponent {
       if (rowIndex > 0 && name === markerTimingRows[rowIndex - 1].name) {
         continue;
       }
+      const displayedName =
+        name.length < MARKER_LABEL_MAX_LENGTH
+          ? name
+          : name.slice(0, MARKER_LABEL_MAX_LENGTH) + '…';
```

Comment:
Non interval markers can have very long names because they embed all the useful data in a string. (eg: `Bailout_TypeBarrierO after getelem on line 159 of https://en.wikipedia.org/w/load.php?debug=false&lang=en&modules=jquery%2Cmediawiki%7Cmediawiki.legacy.wikibits&only=scripts&skin=vector&version=10b6ayp:158`).